### PR TITLE
Improve fmt

### LINKED ISSUE #1: Constructor Settings::style
Some popular styles:

```bnf
use_alias = !{
    String ~ As ~ SYMBOL
  | SYMBOL ~ (ModuleSplit ~ SYMBOL)* ~ As ~ SYMBOL
}
```

```bnf
use_alias = !{
    String ~ As ~ SYMBOL  |
    SYMBOL ~ (ModuleSplit ~ SYMBOL)* ~ As ~ SYMBOL
}
```

```bnf
use_alias = !
{ String ~ As ~ SYMBOL 
| SYMBOL ~ (ModuleSplit ~ SYMBOL)* ~ As ~ SYMBOL
}
```

### LINKED ISSUE #2: instructions  for install and running.
Dearest Maintainer,

In the readme it would be nice to know how it is expected to be installed and ran.  I tried via cargo install first but it says there is nothing.  If nothing else it would prevent a new person from fumbling around.

Thank you for your consideration,

Dictated but not reviewed,
Becker

### PR DESCRIPTION
**Pull request by huacnlee**
## What's News

1. Remove all optional settings to change as built-in rules (Pest grammar is not complex, I make this change to let fmt in simple).
2. Change rule to add space for `{}`, before: `{"a"}`, after: `{ "a" }`.
3. Change rule to add space between `|`, before: `"a"|"b"`, after: `"a" | "b"`.
4. Update `grammar.pest` to newest from Pest project.
5. To supports `LineDoc`, `GrammarDoc`, `BlockComment`.
6. Remove `pre_build.rs`, instead of to use Pest derive.
7. Complete the CLI features to full supports ignore files.
    Now we install by `cargo install pest_fmt`, and then use `pestfmt .` to fix pest files.
9. Allows keeping existing newlines into an empty newline.

To fix #2, #3, and close #1.

### CODE REVIEW

**Comment by tomtau** [comment]
@oovm @sbeckeriv could you help to add the https://github.com/orgs/pest-parser/teams/pest-ide-tools team as owners to the [pest_fmt](https://crates.io/crates/pest_fmt) crate? https://doc.rust-lang.org/cargo/reference/publishing.html#cargo-owner

```
cargo owner --add github:pest-parser/pest-ide-tools
```

**Comment by oovm** [comment]
@tomtau 

```yaml
error: failed to invite owners to crate `pest_fmt` on registry at https://crates.io

Caused by:
  the remote server responded with an error: missing github team argument; format is github:org:team
Caused by:
  the remote server responded with an error: could not find the github team pest-parser/pest-ide-tools
```

image

**Comment by tomtau** [comment]
@oovm how about 
```
cargo owner --add github:pest-parser:pest-ide-tools
```
?

**Comment by oovm** [comment]
Seems some bugs of `crates.io`.

I confirmed that the following users exist, but `crates.io` reports that it cannot be found.

```js
github:pest-parser:pest-ide-tools
github:pest-parser:admins
```

Failed in both command line and web.

image

**Comment by tomtau** [comment]
@oovm maybe it's a permission issue? https://doc.rust-lang.org/cargo/reference/publishing.html#github-permissions

**Comment by tomtau** [comment]
https://doc.rust-lang.org/cargo/reference/publishing.html#troubleshooting-github-team-access-errors

**Comment by tomtau** [comment]
@oovm I got the invitation from you and I added the team there, so it should be fixed now, thanks!

File: Cargo.toml
Comment (tomtau) [inline Cargo.toml:8]:
```suggestion
version = "0.2.1"
```

**Review by tomtau — COMMENTED** [review]
haven't fully reviewed yet, but ok to merge if this is blocking https://github.com/pest-parser/pest-ide-tools/pull/3

**Comment by Jamalam360** [comment]
Not blocking, I'm pretty focused on getting the LSP working for other editors right now. Will take a look at this over the weekend

File: Cargo.toml
Comment (huacnlee) [reply Cargo.toml]: Done

**Comment by huacnlee** [comment]
I found there have a bug will lose `PUSH`, I am going to fix it. Please wait to release.

**Comment by huacnlee** [comment]
https://github.com/huacnlee/autocorrect/commit/c1e57cefb21bfd383c17273c1f1b4c59b1f7e301

I have use pest-fmt to format my project, it all test passed. So there are looks like no missing chars.

File: Readme.md
Comment (tomtau) [inline Readme.md:4]: this is still live, but I guess it won't be updated -- or @sbeckeriv do you plan to maintain the online tool?

File: src/grammar.pest
Comment (tomtau) [inline src/grammar.pest:9]: I'm thinking whether there's a way to have this without copy-pasting the meta grammar?

**Review by tomtau — APPROVED** [review]
had a quick look, perhaps can try to test publishing it?

File: Readme.md
Comment (huacnlee) [reply Readme.md:4]: After this PR merged, we can implement format on pest site by use WASM.

File: src/grammar.pest
Comment (huacnlee) [reply src/grammar.pest:9]: I will find a way for that later.

There is nothing must to do, so I think we can merge and publish a version now.

**Comment by huacnlee** [comment]
I have fixed, But I forgot push.

**Comment by huacnlee** [comment]
I have found a bad case, wait for me to fix.

**Pull request merged**
